pkg/db/db.go

```diff
@@ -1667,6 +1667,54 @@ func (h *DBHandler) DBSelectLatestDeployment(ctx context.Context, tx *sql.Tx, ap
 	return processDeployment(rows)
 }
 
+func (h *DBHandler) DBSelectAllLatestDeploymentsForApplication(ctx context.Context, tx *sql.Tx, appName string) (map[string]Deployment, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBSelectAllLatestDeployments")
+	defer span.Finish()
+
+	selectQuery := h.AdaptQuery(
+		`
+	SELECT
+		deployments.eslVersion,
+		deployments.created,
+		deployments.appname,
+		deployments.releaseVersion,
+		deployments.envName,
+		deployments.metadata
+	FROM (
+		SELECT
+	MAX(eslVersion) AS latest,
```

Comment:
we just observed this to be a rather slow query.
Which is fine for a new unused endpoint, just a note that we need to test the performance of this before rolling out.
I would suggest to make the `max` into a `order by` with `limit 1` (but we need to test the performance either way)